core/src/main/java/org/apache/iceberg/events/CreateSnapshotEvent.java

```diff
@@ -19,21 +19,44 @@
 package org.apache.iceberg.events;
 
 import java.util.Map;
+import org.apache.iceberg.Snapshot;
 
 public final class CreateSnapshotEvent {
   private final String tableName;
+  private final long timestampMillis;
   private final String operation;
   private final long snapshotId;
   private final long sequenceNumber;
   private final Map<String, String> summary;
 
+  public CreateSnapshotEvent(String tableName, Snapshot snapshot) {
```

Comment:
Does this need to be public? Why update the event at all if the purpose is to add the timestamp to `CommitReport`? This is an older API that predates the report, which includes metrics.

Reply:
Yes, for this to be used in `MergingSnapshotProducer` and other places.
It looks we send `CommitReport` via `CreateSnapshotEvent` currently

https://github.com/apache/iceberg/blob/d44cba57c859fef04eec1dcc26838b6331c33702/core/src/main/java/org/apache/iceberg/SnapshotProducer.java#L516-L529